src/IECoreMaya/SceneShapeInterface.cpp

```diff
@@ -1470,3 +1494,79 @@ SceneInterface::Path SceneShapeInterface::fullPathName( std::string relativeName
 	return fullPath;
 }
 
+bool SceneShapeInterface::readConvertParam( CompoundParameterPtr parameters, int attrIndex ) const
+{
+	// Read convert parameter from "queryConvertParameters" attribute and set it to parameters inplace.
+
+	IECorePython::ScopedGILLock gilLock;
+
+	// Create a Python list that has the tokenized string of the attribute value for parsing.
+	boost::python::list parserArgList;
+	{
+		MPlug pConvertParamQueries( thisMObject(), aConvertParamQueries );
+		MPlug pConvertParamQuery = pConvertParamQueries.elementByLogicalIndex( attrIndex );
+		MString paramsStr;
+		pConvertParamQuery.getValue( paramsStr );
+
+		const std::string pstr = paramsStr.asChar();
+		boost::tokenizer<boost::char_separator<char> > t( pstr, boost::char_separator<char>( " " ) );
+		boost::tokenizer<boost::char_separator<char> >::const_iterator it, endIt;
+		for ( it = t.begin(), endIt = t.end(); it != endIt; ++it )
+		{
+			parserArgList.append( *it );
+		}
+	}
+
+	boost::python::dict pyExecLocalsDict;
+	pyExecLocalsDict[ "parserArgList" ] = parserArgList;
+	pyExecLocalsDict[ "parameters" ] = parameters;
+
+	boost::python::object sysModules = boost::python::object( boost::python::handle<>( boost::python::borrowed( PyImport_GetModuleDict() ) ) );
```

Comment:
@johnhaddon do you know if there's a more succinct way to call into python? I can't remember, but I had a feeling that we'd wrapped this up somehow... maybe I'm wrong though.